docs/technical-design/resovler-design.md

```diff
@@ -0,0 +1,146 @@
+      protobuf2[Protobuf] -- HealthPlanPackageFormDataType --> domainToHPPType
+
+      emailNotification -- Domain Model Type --> emailerService
+      emailerService -- Void OR Error --> emailNotification
+      emailerService -- API Request --> sesEmailer[AWS SES Email Service]
+      sesEmailer[AWS SES Email Service] -- API Response --> emailerService
+    end
+    subgraph response[GraphQL Response]
+    end
+    
+    request -- GraphQL Type --> container
+    container -- GraphQL Type --> response
+
+    style request stroke:#3498DB,stroke-width:4px
+    style response stroke:#3498DB,stroke-width:4px
+    style resolver stroke:#3498DB,stroke-width:4px,fill:#1f1f1f
+```
+## General Guidance
+### Postgres Database Operations
+[Postgres](https://www.postgresql.org/docs/) database operations are contained in functions that we call **postgres handlers**. You can find all the handlers in [`services/app-api/src/postgres/`](../../services/app-api/src/postgres). In these handler functions the database operations are performed using [Prisma ORM](https://www.prisma.io/docs/orm) which itself uses Prisma generated types and returns data from the database as these types. So in most of the handlers, data from the operation must be converted to a domain model type before returning to the resolver and errors occurring in the handler functions should be returned to the resolver to handle.
+
+It's important that data from the DB is converted to the domain model before returning to the resolver to adhere to our strategy of using the domain model as the internal communication protocol for different parts of our app. 
+
+Resolvers are passed Postgres handlers via [dependency injection](design-patterns.md#dependency-injection). The configuration for the Postgres handler dependency and many others are done in [apollo_gql.ts](../../services/app-api/src/handlers).
+
+The diagram below is the data flow diagram for `createHealthPlanPackage` resolver.
+```mermaid
+---
+title: createHealthPlanPackage Data Flow Diagram
+---
+flowchart RL
+    subgraph resolver[Resolvers]
+      subgraph createHealthPlanPackage[createHealthPlanPackage]
+        callInsertContract[Insert Contract]
+      end 
+    end
+    
+    subgraph postgresHandlers[Postgres Handlers]
+      subgraph insertContractHandler[insertDraftContract - Handler]
+        prismaOrm[Prisma ORM]
+        parseContractWithHistory[parseContractWithHistory]
+
+        prismaOrm -- Prisma Model --> parseContractWithHistory
+      end 
+    end
+    
+    subgraph dbContainer[Database]
+      postgresDB[(PostgresDB)]
+    end
+
+    callInsertContract -- Domain Model --> insertContractHandler
+    parseContractWithHistory -- Domain Model OR Error --> callInsertContract
+
+    prismaOrm -- Operation --> dbContainer
+    dbContainer -- Data --> prismaOrm
+
+    style resolver stroke-width:4px,fill:#1f1f1f;
+    style postgresHandlers stroke-width:4px,fill:#1f1f1f;
+```
```

Comment:
Really like the granularity of this diagram. Feels right sized. I can digest it. Also like that its showing something that is a bit hard to see in the code without opening multiple files.